src/cpp/shared/combat/combat_spells.cpp

```diff
@@ -1336,34 +1336,42 @@ int __fastcall HandleCastSpell(tag_message &evt) {
   }
 }
 
-void combatManager::Fireball(int hexIdx, int spell) {
-  if(!ValidHex(hexIdx))
-    return;
+void combatManager::AreaSpellMessage(Spell spell, long damage) {
+  switch(spell) {
+    case SPELL_FIREBALL:
+      sprintf(gText, "The fireball does %d damage.", damage);
+      break;
+    case SPELL_FIREBLAST:
+      sprintf(gText, "The fireblast does %d damage.", damage);
+      break;
+    case SPELL_COLD_RING:
+      sprintf(gText, "The cold ring does %d damage.", damage);
+      break;
+    case SPELL_PLASMA_CONE:
+      sprintf(gText, "The plasma cone stream does %d damage.", damage);
+      break;
+    case SPELL_FIRE_BOMB:
+      sprintf(gText, "The fire bomb does %d damage.", damage);
+      break;
+    case SPELL_IMPLOSION_GRENADE:
+      sprintf(gText, "The implosion grenade does %d damage.", damage);
+      break;
+    default:
+      sprintf(gText, "Area spell %d does %d damage.", spell, damage);
+      break;
+  }
+  this->CombatMessage(gText, 1, 1, 0);
+}
 
+void combatManager::AreaSpellDrawImpact(int hexIdx, std::string &iconFileName, int numSprites, bool flip) {
```

Comment:
Not all are used for some reason, so that's why
I didn't want to pollute the code base with an enum that is only used in one function, that's why I didn't make one.
Should I make one nonetheless?

Reply:
Yes. More precise types are the answer to a lot of problems. Don't be afraid to make lots of enums.

Could you say more about numsprites?

Reply:
For example Cold Ring has 10 frames inside ICN file, but only 7 are used when casting it.